lib/spack/spack/cmd/buildcache.py

```diff
@@ -376,9 +387,12 @@ def createtarball(args):
     # restrict matching to current environment if one is active
     env = ev.get_env(args, 'buildcache create')
 
-    _createtarball(env, args.spec_yaml, args.specs, args.directory,
-                   args.key, args.no_deps, args.force, args.rel, args.unsigned,
-                   args.allow_root, args.no_rebuild_index)
+    add_spec = ('package' in args.things_to_install)
+    add_deps = ('dependencies' in args.things_to_install)
```

Comment:
Minor nit:  `args.things_to_install` is just a string literal, right?   Maybe just do a string literal compare with `==` rather than a string subset check with `in`.  In my initial reading, I was thinking `args.things_to_install` was a list (or some other iterable) due to the use of `in`.  But I don't see an `nargs` in the argparse line, so I think it's just a string literal.

Reply:
Yes, `args.things_to_install` is just a string literal, but can take on the compound value `'package,dependencies'` in the default case, which is the reason for the substring check.  This logic is also lifted straight from the `spack install` codepath.  I think it's best as-is since the strings involved are not long enough to introduce performance problems and this logic is future-proof against the more obvious ways we might extend this option.

For example, if we ever decided to explicitly allow the default on the command line, or if we ever add another option, like `'build_dependencies'`, we'd need to be able to handle options like `--only a,b,c` and every permutation thereof.

Reply:
> but can take on the compound value 'package,dependencies' in the default case

Ah! I totally missed that.  Makes total sense now.  Thanks for the explanation!